vavr/src/main/java/io/vavr/collection/Traversable.java

```diff
@@ -860,12 +854,7 @@ default T last() {
      */
     @SuppressWarnings("unchecked")
     default Option<T> min() {
-        if (isEmpty()) {
-            return Option.none();
-        } else {
-            final Traversable<T> ts = isTraversableAgain() ? this : toStream();
-            return ts.minBy(Comparators.naturalComparator());
-        }
+        return minBy(Comparators.naturalComparator());
```

Comment:
* we do not need to wrap a traversable-once into a Stream because no element is touched before reduction
* minBy API say that the empty case is already covered.